pinentryedittext/src/main/java/com/alimuzaffar/lib/pin/PinEntryEditText.java

```diff
@@ -220,21 +220,21 @@ protected void onSizeChanged(int w, int h, int oldw, int oldh) {
             mLastCharPaint.setColor(mOriginalTextColors.getDefaultColor());
             mCharPaint.setColor(mOriginalTextColors.getDefaultColor());
         }
-        int availableWidth = getWidth() - getPaddingRight() - getPaddingLeft();
+        int availableWidth = getWidth() - ViewCompat.getPaddingEnd(this) - ViewCompat.getPaddingStart(this);
         if (mSpace < 0) {
             mCharSize = (availableWidth / (mNumChars * 2 - 1));
         } else {
             mCharSize = (availableWidth - (mSpace * (mNumChars - 1))) / mNumChars;
         }
         mLineCoords = new RectF[(int) mNumChars];
         mCharBottom = new float[(int) mNumChars];
-        int startX = getPaddingLeft();
+        int startX = ViewCompat.getPaddingStart(this);
         int bottom = getHeight() - getPaddingBottom();
         int rtlFlag = 1;
         final boolean isLayoutRtl = TextUtilsCompat.getLayoutDirectionFromLocale(Locale.getDefault()) == ViewCompat.LAYOUT_DIRECTION_RTL;
         if (isLayoutRtl) {
             rtlFlag = -1 * rtlFlag;
-            startX = (int) (getWidth() - getPaddingRight() - mCharSize);
+            startX = (int) (getWidth() - ViewCompat.getPaddingEnd(this) - mCharSize);
```

Comment:
This is now incorrect, paddingStart for RTL is the right side so it should still be width - paddingStart.
Ideally, we should support both, if paddingStart is not set but paddingLeft is, we should use it. For now, just make it work with start and end or revert out to working with left and right and I can add the missing functionality.